src/messenger/messenger.cpp

```diff
@@ -364,32 +374,29 @@ ConnectionId Messenger::MakeConnection(const ConnectionDestination &target) {
   return ConnectionId(common::ManagedPointer(this), target, identity_);
 }
 
-void Messenger::SendMessage(common::ManagedPointer<ConnectionId> connection_id, std::string message, CallbackFn fn) {
-  uint64_t message_id = message_id_++;
+void Messenger::SendMessage(common::ManagedPointer<ConnectionId> connection_id, const std::string &message,
```

Comment:
This function needs to be made thread-safe as well?

Reply:
Added latch to unordered_map, made int atomic, but I think we don't need to do it for the socket -- see comment.